### DISCUSSION TITLE
Substring for Japanese characters

### DISCUSSION THREAD

**Discussion by natarajank11** [trinodb/trino#17719] [General]
In Substring, every character is counted as one byte,
`select substring('全角文字列全角文字列全角文字列全角文字列全角文字列全角文字列全角文字列全角文字列全',1,20);`
Result is,
`全角文字列全角文字列全角文字列全角文字列`

My expectation to result in double byte character, should be `全角文字列全角文字列`

**Comment by ebyhr** [comment]
There're two char length units (CHARACTERS, OCTETS) for substring in SQL standard, but we don't support octets option at this time. 

You can cast to varbinary as the workaround:
```sql
SELECT from_utf8(substr(varbinary '全角文字列全角文字列全角文字列全角文字列全角文字列全角文字列全角文字列全角文字列全',1,21));
     _col0      
----------------
 全角文字列全角 

```
